src/arch/aarch64.rs

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+}
+
+impl AbiArgument for u32 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(32).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i32 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&(*self as u64).to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(32).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u64 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(64).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i64 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 64 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(64).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for u128 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(128).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for i128 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(128).align());
+            buffer.extend_from_slice(&self.to_ne_bytes());
+        }
+    }
+}
+
+impl AbiArgument for Felt {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.to_bytes_le());
+        } else {
+            align_to(buffer, get_integer_layout(252).align());
+            buffer.extend_from_slice(&self.to_bytes_le());
+        }
+    }
+}
+
+impl AbiArgument for U256 {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
+            buffer.extend_from_slice(&self.lo.to_le_bytes());
+            buffer.extend_from_slice(&self.hi.to_le_bytes());
+        } else {
+            align_to(buffer, get_integer_layout(256).align());
+            buffer.extend_from_slice(&self.lo.to_le_bytes());
+            buffer.extend_from_slice(&self.hi.to_le_bytes());
+        }
+    }
+}
+
+impl AbiArgument for [u8; 31] {
+    fn to_bytes(&self, buffer: &mut Vec<u8>) {
+        if buffer.len() < 56 {
```

Comment:
Please add a comment here with the reason why here it checks with 56

Reply:
Done. I also mentioned why `bytes31` also uses the same splitting rules as other integers.